src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php

```diff
@@ -86,6 +86,10 @@ public function get(string $name): array|bool|string|int|float|\UnitEnum|null
 
     public function set(string $name, array|bool|string|int|float|\UnitEnum|null $value)
     {
+        if (is_numeric($name)) {
+            trigger_deprecation('symfony/dependency-injection', '6.2', 'Using numeric parameter names is deprecated and will throw as of 7.0.');
```

Comment:
let's give the name of the offending parameter:
`'Using numeric parameter name "%s" is deprecated and will throw as of 7.0.', $name);`
can you also please add the throwing line commented to make it easy to upgrade on 7.0?

Reply:
Done.